lib/rubocop/cop/style/or_assignment.rb

```diff
@@ -81,7 +81,12 @@ def take_variable_and_default_from_ternary(node)
         end
 
         def take_variable_and_default_from_unless(node)
-          variable, default = *node.if_branch
+          if node.if_branch
+            variable, default = *node.if_branch
```

Comment:
Interesting. I thought this would trigger another cop, that would suggest:

```ruby
variable, default = if node.if_branch
                      *node.if_branch
                    else
                      *node.else_branch
                    end
```

Maybe a false negative in that other cop? 🙂

Reply:
This syntax is a syntax error. I learned at that time :-)

```console
% cat example.rb
variable, default = if node.if_branch
                      *node.if_branch
                    else
                      *node.else_branch
                    end

% ruby -c example.rb
example.rb:2: syntax error, unexpected '\n', expecting '='
example.rb:4: syntax error, unexpected '\n', expecting '='
```

Reply:
Oh. Even more interesting. I only tested using this:

```ruby
a, b = if foo; c; else; d; end
```

which works. 😂

Reply:
I got a syntax error in my environment 🤔 

```console
% rbenv each ruby -e 'a, b = if foo; *c; else; *d; end'

(snip)

SyntaxError: -e:1: syntax error, unexpected ';'
a, b = if foo; *c; else; *d; end
                 ^
FAILED IN: 1.8.7-p375 1.9.3-p551 2.0.0-p648 2.1.10 2.2.10 2.2.2 2.3.5 2.3.7 2.4.1 2.4.6 2.4.7 2.5.0-dev 2.5.0-gcc4 2.5.6 2.6.0-dev 2.6.4 2.7.0-dev jruby-1.7.19 jruby-9.0.5.0 jruby-9.1.17.0 jruby-9.2.7.0 jruby-9.2.8.0
```

Reply:
Strange. Now I'm getting the same thing. 🤔 Maybe need more coffee. ☕️

Reply:
> I thought this would trigger another cop

Ignoring that in this case a syntax error occurs, `Style/ConditionalAssignment` comes close to checking for this, but I think it only looks for a single variable. There is an explicit check that the assignment isn't mass assignment.


```ruby
        def allowed_statements?(branches)
          return false unless branches.all?

          statements = branches.map { |branch| tail(branch) }.compact

          lhs_all_match?(statements) && statements.none?(&:masgn_type?) &&
            assignment_types_match?(*statements)
        end
```

We certainly could add the functionality, but we'll need to remember to not flag splat expansion for correction.

Side note: There are some other "assignment" types that this cop could check for as well. The other "assignment" types are really operators such as `+` or `-`. I think I left them out because of potential confusion over the name of the cop and what it would be recommending. It would be similar to how the cop supports `=~`, `!~`, and `<=>`.